src/settings.rs

```diff
@@ -0,0 +1,82 @@
+use glib::variant::FromVariant;
+use glib::{BoolError, IsA, ToVariant};
+use {Settings, SettingsExt};
+
+pub trait SettingsExtManual {
+    fn get<U: FromVariant>(&self, key: &str) -> U;
+
+    fn set<U: ToVariant>(&self, key: &str, value: &U) -> Result<(), BoolError>;
+}
+
+impl<O: IsA<Settings>> SettingsExtManual for O {
+    fn get<U: FromVariant>(&self, key: &str) -> U {
+        let val = self.get_value(key);
+        FromVariant::from_variant(&val).expect(&format!(
+            "Type mismatch: Expected '{}' got '{}'",
+            U::static_variant_type().to_str(),
+            val.type_()
+        ))
+    }
+
+    fn set<U: ToVariant>(&self, key: &str, value: &U) -> Result<(), BoolError> {
+        self.set_value(key, &value.to_variant())
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use std::env::set_var;
+    use std::process::Command;
+    use std::str::from_utf8;
+
+    fn set_env() {
+        let output = Command::new("glib-compile-schemas")
```

Comment:
Not that it helps, but you should wrap all code in this function in `std::sync::Once` :)

Reply:
Oh nice, hadn't known about that, thought about using a global bool for that at first but that's way prettier, thanks :)

Reply:
Pushed, thanks